app/services/fed_email_domains.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class FedEmailDomains
+  FED_EMAIL_DOMAINS_PATH = Rails.root.join(IdentityConfig.store.fed_domain_file_path).freeze
+
+  def self.email_is_fed_domain?(domain)
+    File.read(FED_EMAIL_DOMAINS_PATH).scan(/#{domain}/).present?
```

Comment:
Re: File reading vs. database table, I'd be curious if there's a noticeable performance difference. For pwned passwords, we use [`BinarySearchSortedHashFile`](https://github.com/18F/identity-idp/blob/main/app/services/binary_search_sorted_hash_file.rb) which is an optimized algorithm for searching in a sorted file. Maybe we can use that?

Reply:
Yea I was curious if we wanted to use a database table, Would that mean we would need to hash the data we get from the data file before we can search through it?

Reply:
If you're talking about `BinarySearchSortedHashFile`, yeah I think so. Either that or abstract out a more generic search class that can operate on plaintext, and maybe `BinarySearchSortedHashFile` can extend that with its hash-specific requirements.

Reply:
I think it could also be reasonable to just expect the callsite to pre-hash the `key` value for pwned passwords, e.g.

```rb
BinarySearchSortedFile.new(file).call(Digest::SHA1.hexdigest(password).upcase)
```

Reply:
What's the rough number of these email addresses?

Reply:
@zachmargolis It's based on https://github.com/cisagov/dotgov-data/blob/main/current-federal.csv

Reply:
Yeah flat file is probably faster that a roundtrip to the DB, but DB is much easier to maintain. And the domains are small that a DB index is probably just fine for them.

I'd be curious about modding the `BinarySearchSortedHashFile` to search a plaintext file but I think it relies on the digest size being identical per row so it can jump to an arbitrary position in the file, so it wouldn't be a very human-readable file

Reply:
> I think it relies on the digest size being identical per row so it can jump to an arbitrary position in the file

Ah yeah, I'd forgot that's part of the search algorithm. We could convert the domains to hashes as part of loading it, though at the expense of readability like you mentioned.

Reply:
I think I will just store in the DB, makes it easier to maintain imo.